packages/api-server/api_server/routes/tasks/tasks.py

```diff
@@ -121,8 +119,102 @@ async def query_task_states(
             if status_string not in valid_values:
                 continue
             filters["status__in"].append(mdl.Status(status_string))
+    return DbTaskState.filter(**filters)
 
-    return await task_repo.query_task_states(DbTaskState.filter(**filters), pagination)
+
+@router.get("/count", response_model=int)
+async def task_states_count(
```

Comment:
I don't think we should have this endpoint, counting the number of rows is an expensive db operation.

Reply:
Hmmm, I didn't know this was the case, thanks for the heads up. I'll do pagination directly

Reply:
https://github.com/open-rmf/rmf-web/pull/909/commits/40b91f0941bff7e7123fa9fb8cf3a284f0c57e86